### ISSUE TITLE
Qwen Img fp16 NaN issue

### ISSUE THREAD

**Issue by realisticdreamer114514** [comfyanonymous/ComfyUI#10751]
### Custom Node Testing

- [x] I have tried disabling custom nodes and the issue persists (see [how to disable custom nodes](https://docs.comfy.org/troubleshooting/custom-node-issues#step-1%3A-test-with-all-custom-nodes-disabled) if you need help)

### Your question

Continuing https://github.com/comfyanonymous/ComfyUI/issues/10668 but a new issue because I'll point out an obvious likely fix right now, need guidance from technically proficient people here

tldr; Qwen img models will produce NaN in latent, which is the cause of black preview/output at unet infer/vae decode stage across models, when running/cast to fp16 on some hardware. In qwen img models it is likely a unet issue. `torch.clamp(65536)` when `--fp16-unet` would be a start to fix this but at which point exactly? 

Again, are there other possible fixes that doesn't involve editing the code base e.g. venv dependency/flags/node usage fixes?

### Logs

```powershell

```

**Comment by realisticdreamer114514** [comment]
No one? 

https://engineering.drawthings.ai/i/172283351/activation-dynamics

**Comment by realisticdreamer114514** [comment]
Find a workaround (not a fix because it involves clamping & so precision loss):
in `\comfy\ldm\qwen_image\model.py`, `class QwenImageTransformerBlock(nn.Module):`, `def forward(`
add
```
        # Clip to prevent overflow for fp16
        if encoder_hidden_states.dtype == torch.float16:
            encoder_hidden_states = encoder_hidden_states.clip(-65504, 65504)
        if hidden_states.dtype == torch.float16:
            hidden_states = hidden_states.clip(-65504, 65504)
```
before `return encoder_hidden_states, hidden_states` to clamp values within fp16 range

Update: Workaround doesn't work even if it should (black imgs should be a NaN in latent problem)

**Comment by Laidawang** [comment]
i think its a problem caused by --force-fp16，My black problem fixed after I removed it.
